commands/connect/mapping-write-errors.js

```diff
@@ -1,37 +1,36 @@
 'use strict'
 const api = require('../../lib/connect/api.js')
 const regions = require('../../lib/connect/regions.js')
+const pg = require('heroku-pg')
 const cli = require('heroku-cli-util')
 const co = require('co')
 
 function * run (context, heroku) {
   context.region = yield regions.determineRegion(context, heroku)
-  let mappingName = context.args.name
+  let mappingName = context.args.name.toLowerCase()
   let connection = yield api.withConnection(context, heroku)
-  let mapping = yield api.withMapping(connection, mappingName)
-  let results = yield cli.action('Retrieving errors', co(function * () {
-    let url = '/api/v3/mappings/' + mapping.id + '/errors'
-    return yield api.request(context, 'GET', url)
-  }))
-  let errors = results.json
+  let query = `
+SELECT
+  id, processed_at, record_id, sf_message, false AS archived
+FROM
+  ${connection.schema_name}._trigger_log
+WHERE
+  state = 'FAILED' AND table_name = '${mappingName}'
+UNION
+SELECT
+  id, processed_at, record_id, sf_message, true AS archived
+FROM
+  ${connection.schema_name}._trigger_log_archive
+WHERE
+  state = 'FAILED' AND table_name = '${mappingName}'
+ORDER BY
+  processed_at DESC
+`
+  let fetcher = pg.fetcher(heroku)
+  let db = yield fetcher.database(context.app, connection.db_key)
 
-  if (errors.count === 0) {
-    cli.log('No write errors in the last 24 hours')
-  } else {
-    if (context.flags.json) {
-      cli.styledJSON(errors.results)
-    } else {
-      cli.table(errors.results, {
-        columns: [
-          {key: 'id', label: 'Trigger Log ID'},
-          {key: 'table_name', label: 'Table Name'},
-          {key: 'record_id', label: 'Table ID'},
-          {key: 'message', label: 'Error Message'},
-          {key: 'created_at', label: 'Created'}
-        ]
-      })
-    }
-  }
+  let output = yield pg.psql.exec(db, query)
+  cli.log(output)
```

Comment:
Does the output of `pg.psql.exec` automatically format nicely when output? Or is there some other reason we're doing away with the styled table?

Reply:
It formats it as a table from PostgreSQL (like if you ran the SQL query in psql yourself). I couldn't find anyone that was parsing the output and using a styled table. It seems like our options are either using our endpoint or using psql directly. I'm not sure which is better honestly